cpp/oneapi/dal/algo/kmeans_init/compute_types.hpp

```diff
@@ -21,12 +21,18 @@
 namespace oneapi::dal::kmeans_init {
 
 namespace detail {
+template <typename Task = task::by_default>
 class compute_input_impl;
+
+template <typename Task = task::by_default>
 class compute_result_impl;
 } // namespace detail
 
+template <typename Task = task::by_default>
 class ONEAPI_DAL_EXPORT compute_input : public base {
```

Comment:
Is it really depend on Task type in theory?

Reply:
should be discussed with @rlnx

Reply:
resolved